src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,461 @@
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        amount = bound(amount, 0, type(uint256).max - prevToBalance);
+        deal(_token, address(this), TokenLike(_token).balanceOf(address(this)) + amount, true);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevSenderBalance = TokenLike(_token).balanceOf(address(this));
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(address(this), to, amount);
+        assertTrue(TokenLike(_token).transfer(to, amount));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        if (address(this) == to) {
+            assertEq(TokenLike(_token).balanceOf(address(this)), prevSenderBalance);
+        } else {
+            assertEq(TokenLike(_token).balanceOf(address(this)), prevSenderBalance - amount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + amount);
+        }
+    }
+
+    function fuzzCheckTransferFrom(
+        address _token,
+        address to,
+        uint256 approval,
+        uint256 amount
+    ) public {
+        vm.assume(to != address(0) && to != _token);
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        approval = bound(approval, 0, type(uint256).max - prevToBalance);
+        amount = bound(amount, 0, approval);
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(from) + amount, true);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+        vm.prank(from);
+        TokenLike(_token).approve(address(this), approval);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, to, amount);
+        assertTrue(TokenLike(_token).transferFrom(from, to, amount));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        uint256 app = from == address(this) || approval == type(uint256).max ? approval : approval - amount;
+        assertEq(TokenLike(_token).allowance(from, address(this)), app);
+        if (from == to) {
+            assertEq(TokenLike(_token).balanceOf(from), prevFromBalance);
+        } else  {
+            assertEq(TokenLike(_token).balanceOf(from), prevFromBalance - amount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + amount);
+        }
+    }
+
+    function fuzzCheckTransferInsufficientBalance(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount,
+        uint256 sendAmount
+    ) public {
+        vm.assume(to != address(0) && to != address(_token));
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply - 1);
+        sendAmount = bound(sendAmount, mintAmount + 1, type(uint256).max);
+        deal(_token, address(this), TokenLike(_token).balanceOf(address(this)) + mintAmount, true);
```

Comment:
`sendAmount` can be <= than the balance if there was any prev one.
In this case I think the easiest way to solve this is:
```suggestion
        deal(_token, address(this), mintAmount, true);
```